cityofaustin/atd-moped
# Adds basic user analytics tracking

## Associated issues
- https://github.com/cityofaustin/atd-data-tech/issues/4839

This PR adds user activity logging by making use of a [custom postgres function](https://hasura.io/docs/latest/schema/postgres/custom-functions/) that is exposed as a mutation through the Hasura API. When a logged-in user mutates this endpoint, the user's ID is logged to a `moped_user_events` table.

This mutation is called automatically when the Moped Editor `<App>` component mounts, thereby providing basic metrics about who is using Moped.

This function also updates a new column on the `moped_users` table called `last_seen_date`. This column can be viewed on the Staff list in the Editor.

## Testing
**URL to test:** Local

1. Login and navigate to the staff page. Notice the **Last seen** column should have a recent timestamp for your user account. Nice!
2. Refresh your page—the **Last seen** column should update accordingly. It may take a second refresh for this to stick. I think there may be a race condition going on there, since it is not certain that the `moped_users` query is going to resolve after  your `last_seen_date` is updated.
3. Without refreshing the page, navigate around the app, clicking on random things, until eventually coming back to the staff list. Your **Last seen** timestamp should not have changed.
4. Use your SQL client to inspect the `moped_user_events` table. It is accumulating one event every time you refresh your page. If you log out of Moped it will also log an event if you log back in.




---
#### Ship list
